### ISSUE TITLE
RC5 - Unable to locate .truck file glitch

### ISSUE THREAD

**Issue by DiggerFan13** [RigsOfRods/rigs-of-rods#2429]
While running on RC5, I attempted to spawn a monster truck, Bear Foot. It spawned as normal with no issue, I then went to spawn Carolina Crusher (a truck from the same pack as Bear Foot and cloned from Bear Foot so there are no conflicts with any materials/meshes. They are entirely separate trucks with separate .truck files, etc.) it said "Unable to locate .truck file"

  Now that's just silly because it wouldn't show up on the list if it wasn't there to begin with. I thought maybe it was a glitch or something, so I closed out and reloaded. The same issue happened as before, no change. I also tried despawning Bear Foot first, no change. I then reloaded the game and opened a map fresh and spawned Carolina Crusher. I then attempted to spawn Bear Foot, however it said the .truck file could not be located again. So basically, once one is spawned, the other can't/won't spawn unless you close out of the game and load it back up WITHOUT the other vehicle spawned first. This issue is NOT isolated to just these trucks (they were just my example), it's anything that's been cloned it seems. 

  This issue did not exist prior to RC5 as far as I'm aware, I can spawn both just fine in RC4. I included an image of them both spawned in RC4. I am also including a link to the download of the trucks: http://www.mediafire.com/file/qv35bqum63z4p8l/BearFootChevy%2526CarolinaCrusherCustomBreakable.zip/file

![BFCC](https://user-images.githubusercontent.com/54759914/64071788-c587ba00-cc4f-11e9-875c-66449ea580a8.png)

**Comment by CuriousMike56** [comment]
Hmm..both trucks load fine here:
![image](https://user-images.githubusercontent.com/46073351/64072254-90806500-cc59-11e9-88b4-76b7dc250314.png)
Could you upload your RoR.log (after reproducing the issue)?

**Comment by DiggerFan13** [comment]
So upon reinstalling RC5, now they spawn for me as well. However, I know the issue still exists as a friend of mine still has it, so I had him replicate the issue and send me his log. Attached is his log, you can see he tried to spawn Carolina Crusher second but it says the .truck file is missing. 
[RoR.log](https://github.com/RigsOfRods/rigs-of-rods/files/3563145/RoR.log)

**Comment by CuriousMike56** [comment]
Figured out the problem:
```
02:10:27: Ogre::ItemIdentityException::ItemIdentityException: ParticleSystem template with name 'tracks/smokeblz' already exists. in ParticleSystemManager::createTemplate at C:\conan\OGRE\1.11.6\anotherfoxguy\stable\build\ca009a9d3962c6e22e649bbe37d4642d4064b682\OgreMain\src\OgreParticleSystemManager.cpp (line 192)
02:10:27: [RoR] Error while loading 'C:\Users\NCG User\Documents\Rigs of Rods 0.4\vehicles\CarolinaCrusherCustomBreakable.zip', message: Ogre::ItemIdentityException::ItemIdentityException: ParticleSystem template with name 'tracks/smokeblz' already exists. in ParticleSystemManager::createTemplate at C:\conan\OGRE\1.11.6\anotherfoxguy\stable\build\ca009a9d3962c6e22e649bbe37d4642d4064b682\OgreMain\src\OgreParticleSystemManager.cpp (line 192)
.....
02:10:27: [RoR] Failed to load 'CarolinaCrusherCustomBreakable.truck' (type: 'actor'), message: Truckfile not found
```
Ogre fails to load a particle which causes RoR to unload the zip file, thus throwing the "not found" message. This should be fixed on master (https://github.com/RigsOfRods/rigs-of-rods/pull/2398).
For RC5, removing `smoke.particle` from the truck's zip should fix it.

**Comment by DiggerFan13** [comment]
So the issue has since resurfaced for me as well. This time, I was unable to spawn Monster Patrol, which is also cloned off the same base as the other two. I had both Bear Foot and Carolina Crusher spawned already and tried spawning Monster Patrol Retro Custom but got the missing .truck file message. I've also included a screenshot of the error message. Here is the link to the truck: http://www.mediafire.com/file/ck80u8lqva8ksnj/MonsterPatrolCustomBreakable.zip/file

![Glitch](https://user-images.githubusercontent.com/54759914/64072665-b2c9b100-cc60-11e9-9601-5ba75da3a77e.png)

[RoR.log](https://github.com/RigsOfRods/rigs-of-rods/files/3563161/RoR.log)

**Comment by DiggerFan13** [comment]
> Figured out the problem:
> 
> ```
> 02:10:27: Ogre::ItemIdentityException::ItemIdentityException: ParticleSystem template with name 'tracks/smokeblz' already exists. in ParticleSystemManager::createTemplate at C:\conan\OGRE\1.11.6\anotherfoxguy\stable\build\ca009a9d3962c6e22e649bbe37d4642d4064b682\OgreMain\src\OgreParticleSystemManager.cpp (line 192)
> 02:10:27: [RoR] Error while loading 'C:\Users\NCG User\Documents\Rigs of Rods 0.4\vehicles\CarolinaCrusherCustomBreakable.zip', message: Ogre::ItemIdentityException::ItemIdentityException: ParticleSystem template with name 'tracks/smokeblz' already exists. in ParticleSystemManager::createTemplate at C:\conan\OGRE\1.11.6\anotherfoxguy\stable\build\ca009a9d3962c6e22e649bbe37d4642d4064b682\OgreMain\src\OgreParticleSystemManager.cpp (line 192)
> .....
> 02:10:27: [RoR] Failed to load 'CarolinaCrusherCustomBreakable.truck' (type: 'actor'), message: Truckfile not found
> ```
> 
> Ogre fails to load a particle which causes RoR to unload the zip file, thus throwing the "not found" message. This should be fixed on master (#2398).
> For RC5, removing `smoke.particle` from the truck's zip should fix it.

Ah, alright, I'll remove that. I don't even know why it's there to begin with.

**Comment by AnotherFoxGuy** [comment]
Seems to have already been fixed on master
